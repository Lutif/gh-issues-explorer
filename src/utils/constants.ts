export const PAGE_SIZE = 10;
export const REACT_APP_GITHUB_BASE_URL = "https://api.github.com/graphql";
