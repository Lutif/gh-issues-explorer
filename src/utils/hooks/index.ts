export * from "./useSearchIssue";
export * from "./useFetchIssueDetails";
