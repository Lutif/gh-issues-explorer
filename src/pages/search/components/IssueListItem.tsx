import React from "react";
import { Issue, IssueState } from "../../../utils";
import moment from "moment";
import { useHistory, useLocation } from "react-router-dom";
import { IssueListItemWrapper } from "./issueListItem.styled";

interface IssueListItemProps {
  issue: Issue;
}

export const IssueListItem: React.FC<IssueListItemProps> = ({ issue }) => {
  const {
    title,
    state,
    createdAt,
    author: { login } = {},
    comments: { totalCount: totalComments } = {},
  } = issue;
  const history = useHistory();
  const { pathname } = useLocation();

  const gotoIssueDetail = () => history.push(`${pathname}/${issue.number}`);

  const content = `#${issue.number} ${"Opened"} ${moment(
    createdAt
  ).fromNow()} by ${login}`;

  return (
    <IssueListItemWrapper onClick={gotoIssueDetail}>
      <div className="title">{title}</div>
      <div className="details">{content}</div>
      <div>
        {totalComments || 0} comments | {state}{" "}
      </div>
    </IssueListItemWrapper>
  );
};
