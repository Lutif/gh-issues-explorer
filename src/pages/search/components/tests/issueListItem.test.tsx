import { render, screen } from "@testing-library/react";
import { IssueState } from "../../../../utils";
import { IssueListItem } from "../IssueListItem";
import { MemoryRouter as Router } from "react-router-dom";
import { genrateTestIssue } from "./helpers";
import moment from "moment";

const testIssue = genrateTestIssue();
describe("<IssueListItem /> tests", () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it("renders the IssueListItem component", () => {
    render(
      <Router>
        <IssueListItem issue={testIssue} />
      </Router>
    );
    const element = screen.getByText("test");
    expect(element).toBeInTheDocument();
  });

  it("renders the IssueListItem component with closed issue", () => {
    render(
      <Router>
        <IssueListItem issue={{ ...testIssue, closed: true }} />
      </Router>
    );
    const element = screen.getByText("test");
    expect(element).toBeInTheDocument();
  });

  it("test issueListItem snapshot", () => {
    const { asFragment } = render(
      <Router>
        <IssueListItem issue={testIssue} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("test issueListItem content match", () => {
    const content = `#${testIssue.number} ${"Opened"} ${moment(
      testIssue.createdAt
    ).fromNow()} by ${testIssue.author.login}`;
    render(
      <Router>
        <IssueListItem issue={testIssue} />
      </Router>
    );
    const element = screen.getByText(content);
    expect(element).toBeInTheDocument();
  });
});
