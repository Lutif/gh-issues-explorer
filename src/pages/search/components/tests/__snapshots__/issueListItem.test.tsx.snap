// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssueListItem /> tests test issueListItem snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ eyycDj"
  >
    <div
      class="title"
    >
      test
    </div>
    <div
      class="details"
    >
      #2 Opened a few seconds ago by test
    </div>
    <div>
      1 comments | OPEN 
    </div>
  </div>
</DocumentFragment>
`;
